apis/workloads/v1alpha1/replicatedstatemachine_types.go

```diff
@@ -30,18 +30,38 @@ import (
 
 // ReplicatedStateMachineSpec defines the desired state of ReplicatedStateMachine
 type ReplicatedStateMachineSpec struct {
-	// Replicas defines number of Pods
+	// replicas is the desired number of replicas of the given Template.
+	// These are replicas in the sense that they are instantiations of the
+	// same Template, but individual replicas also have a consistent identity.
+	// If unspecified, defaults to 1.
 	// +kubebuilder:default=1
 	// +kubebuilder:validation:Minimum=0
 	// +optional
-	Replicas int32 `json:"replicas,omitempty"`
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
Does the data replication been supported transparently at scaling-out?

Reply:
not in this PR.
and I prefer to respect `datasourceRef` to provide data replication, rather than do it in RSM.
will discuss this in other PR.